py/client/pydeephaven/_table_interface.py

```diff
@@ -195,56 +197,57 @@ def tail(self, num_rows: int):
         table_op = TailOp(num_rows=num_rows)
         return self.table_op_handler(table_op)
 
-    def natural_join(self, table: Any, on: List[str], joins: List[str] = []):
+    def natural_join(self, table: Any, on: Union[str, List[str]], joins: Union[str, List[str]] = None):
         """ Perform a natural-join between this table as the left table and another table as the right table) and
         returns the result table.
 
         Args:
             table (Table): the right-table of the join
-            on (List[str]): the columns to match, can be a common name or an equal expression,
+            on (Union[str, List[str]]): the column(s) to match, can be a common name or an equal expression,
                 i.e. "col_a = col_b" for different column names
-            joins (List[str], optional): a list of the columns to be added from the right table to the result
-                table, can be renaming expressions, i.e. "new_col = col"; default is empty
+            joins (Union[str, List[str]], optional): the column(s) to be added from the right table to the result
+                table, can be renaming expressions, i.e. "new_col = col"; default is None
```

Comment:
On the `joins` for the join methods, we are not clear about what the default does.